src/runtime_src/core/common/info_aie.cpp

```diff
@@ -63,6 +63,222 @@ addnodelist(const std::string& search_str, const std::string& node_str,
   output_pt.add_child(node_str, pt_array);
 }
 
+void 
+populate_bd_info(const boost::property_tree::ptree& input_pt, boost::property_tree::ptree& pt_bd,
+	          std::vector<std::string>& bd_info) {
+  try {
+    boost::property_tree::ptree bd_info_array;
+    const boost::property_tree::ptree& bd_node = input_pt.get_child("bd");
+    
+    for (const auto& pair : bd_node) {
+      const std::string& key = pair.first;
```

Comment:
you may need to add a error condition. bd_info size and value_tree size should match,